### ISSUE TITLE
Decrypting a compressed JWT consistently fails 

### ISSUE THREAD

**Issue by hagmonk** [funcool/buddy-core#30]
Hopefully I'm doing something wrong here :) I was having trouble decrypting a JWT that was encrypted via https://github.com/jwt/ruby-jwe ... however I managed to get the same error to occur directly with buddy:

```
;; openssl genrsa 2048 > private-foobar.pem
;; openssl rsa -in private.pem -out public.pem -outform PEM -pubout
(-> {:a "b"}
    (buddy.sign.jwt/encrypt (keys/public-key "public-foobar.pem") {:alg :rsa-oaep :enc :a128cbc-hs256})
    (buddy.sign.jwt/decrypt (keys/private-key "private-foobar.pem") {:alg :rsa-oaep :enc :a128cbc-hs256}))
```

This returns the hash back successfully.

Adding `:zip true` causes the following failure:

```
1. Unhandled java.lang.ClassCastException
   clojure.core$bytes cannot be cast to [B

                      REPL:   38  buddy.util.deflate/uncompress
                      REPL:   38  buddy.util.deflate/uncompress
                      REPL:   80  buddy.sign.jwe/decode-payload
                      REPL:   77  buddy.sign.jwe/decode-payload
                      REPL:  252  buddy.sign.jwe/decrypt
                      REPL:  232  buddy.sign.jwe/decrypt
                   jwt.clj:   97  buddy.sign.jwt/decrypt
                   jwt.clj:   93  buddy.sign.jwt/decrypt
                      REPL:   78  blip-api.authentication/eval25585
                      REPL:   76  blip-api.authentication/eval25585
             Compiler.java: 6927  clojure.lang.Compiler/eval
             Compiler.java: 6890  clojure.lang.Compiler/eval
                  core.clj: 3105  clojure.core/eval
                  core.clj: 3101  clojure.core/eval
                  main.clj:  240  clojure.main/repl/read-eval-print/fn
                  main.clj:  240  clojure.main/repl/read-eval-print
                  main.clj:  258  clojure.main/repl/fn
                  main.clj:  258  clojure.main/repl
                  main.clj:  174  clojure.main/repl
               RestFn.java:  137  clojure.lang.RestFn/applyTo
                  core.clj:  646  clojure.core/apply
                  core.clj:  641  clojure.core/apply
                regrow.clj:   18  refactor-nrepl.ns.slam.hound.regrow/wrap-clojure-repl/fn
               RestFn.java: 1523  clojure.lang.RestFn/invoke
    interruptible_eval.clj:   87  clojure.tools.nrepl.middleware.interruptible-eval/evaluate/fn
                  AFn.java:  152  clojure.lang.AFn/applyToHelper
                  AFn.java:  144  clojure.lang.AFn/applyTo
                  core.clj:  646  clojure.core/apply
                  core.clj: 1881  clojure.core/with-bindings*
                  core.clj: 1881  clojure.core/with-bindings*
               RestFn.java:  425  clojure.lang.RestFn/invoke
    interruptible_eval.clj:   85  clojure.tools.nrepl.middleware.interruptible-eval/evaluate
    interruptible_eval.clj:   55  clojure.tools.nrepl.middleware.interruptible-eval/evaluate
    interruptible_eval.clj:  222  clojure.tools.nrepl.middleware.interruptible-eval/interruptible-eval/fn/fn
    interruptible_eval.clj:  190  clojure.tools.nrepl.middleware.interruptible-eval/run-next/fn
                  AFn.java:   22  clojure.lang.AFn/run
   ThreadPoolExecutor.java: 1142  java.util.concurrent.ThreadPoolExecutor/runWorker
   ThreadPoolExecutor.java:  617  java.util.concurrent.ThreadPoolExecutor$Worker/run
```

This was in my REPL environment which is as follows:

```
;; Connected to nREPL server - nrepl://localhost:51290
;; CIDER 0.13.0snapshot (package: 20160602.809), nREPL 0.2.12
;; Clojure 1.8.0, Java 1.8.0_71
```

**Comment by niwinz** [comment]
seems like a bug, I'll try to fix it ASAP

**Comment by niwinz** [comment]
I have reproduced the bug, and it has easy fix, I'll try to bump a new patch version ASAP.

**Comment by hagmonk** [comment]
Thank you so much for the prompt response!

**Comment by niwinz** [comment]
The bug is properly fixed at buddy-core `0.13.0`. You can force it in your project until new buddy-sign is released with that new version referenced.

**Comment by hagmonk** [comment]
Confirmed resolved in buddy-core 0.13.0. Thanks Andrey!
